ENH: get rid of the defaults file

schacht-certat opened the pull request:
The defaults file was used to define global settings. This commit drops
the defaults file as a separate files and instead uses the 'global' key
in the runtime conf to define settings that should be effective in every
bot.
To accomplish that there is now a utils.get_global_settings() function
that extracts the 'global' dict from the runtime configuration file.

wagner-certat reviewed (changes requested):
`defaults.conf` is still mentioned in the docs.

The file itself is also still there, do you plan to remove that in a separate PR? And then we need an upgrade function to
1) migrate the values (only those which differ them the internal defaults) to the runtime config
2) remove the file

wagner-certat commented:
In https://github.com/certtools/intelmq/blob/01e8d0184b45d5247723844dd43c7d4d8e7b42da/intelmq/bin/intelmqctl.py#L1522 the "global" (non)bot needs to be skipped

schacht-certat commented on intelmq/lib/test.py:
```diff
@@ -73,6 +71,10 @@ def mocked(conf_file):
 
 
+def mocked_get_global_settings():
+    return BOT_CONFIG
```
An empty dict does not work but I've reduced the values as much as possible.

schacht-certat commented:
> `defaults.conf` is still mentioned in the docs.

fixed.

> 
> The file itself is also still there, do you plan to remove that in a separate PR?

fixed.

> And then we need an upgrade function to
> 
>     1. migrate the values (only those which differ them the internal defaults) to the runtime config
> 
>     2. remove the file

I added an upgrade function that removes the file and migrates the values, but I'm not sure how to find out how the values differ from internal defaults- the internal defaults might be distributed over multiple files...

schacht-certat commented:
> In
> 
> https://github.com/certtools/intelmq/blob/01e8d0184b45d5247723844dd43c7d4d8e7b42da/intelmq/bin/intelmqctl.py#L1522
> the "global" (non)bot needs to be skipped

fixed

wagner-certat reviewed (changes requested):
I'm missing docs in changelog (short) and the NEWS file (explanation to users, what happens on upgrade, e.g. values are migrated to runtime to global, user should check if some values are obsolete etc.)

schacht-certat commented on docs/user/configuration-management.rst:
```diff
@@ -61,13 +60,6 @@ Use the IntelMQ Manager mentioned above to generate the configuration files if u
 
 In the shipped examples 4 collectors and parsers, 6 common experts and one output are configured. The default collector and the parser handle data from malware domain list, the file output bot writes all data to ``/opt/intelmq/var/lib/bots/file-output/events.txt``/``/var/lib/intelmq/bots/file-output/events.txt``.
 
-*******************************
```
Fixed

schacht-certat commented on docs/user/configuration-management.rst:
```diff
@@ -298,7 +290,7 @@ The configuration and the differences are outlined here.
 The AMQP support is considered beta, so small problems might occur. So far, only RabbitMQ as broker has been tested.
 
-You can change the broker for single bots (set the parameters in the runtime configuration per bot) or for the whole botnet (in defaults configuration).
+You can change the broker for single bots (set the parameters in the runtime configuration per bot) or for the whole botnet.
```
Fixed

wagner-certat commented on NEWS.md:
```diff
@@ -41,6 +41,9 @@ The old names can still be used in code, and they are automatically converted to
 ### Configuration
 
+The `defaults.conf` file was removed. Settings that should effect all the bots are not part of the runtime.conf file and are configured in the `global` section in that file.
+The `intelmqctl upgrade` command migrates the existing values from the `defaults.conf` file to the `runtime.conf` file under the `global` section and then deletes the `defaults.conf` file.
```
It's `intelmqctl upgrade-config` :)
